src/mbgl/gl/upload_pass.cpp

```diff
@@ -170,21 +170,35 @@ gfx::AttributeBindingArray UploadPass::buildAttributeBindings(
     }
 
     // For each attribute in the program, with the corresponding default and optional override...
-    const auto resolveAttr = [&](const std::string& name, const auto& defaultAttr, const auto& overrideAttr) -> void {
-        const auto& effectiveAttr = overrideAttr ? *overrideAttr : defaultAttr;
-        const auto& effectiveGL = static_cast<const gl::VertexAttributeGL&>(effectiveAttr);
+    const auto resolveAttr = [&](const std::string& name, auto& defaultAttr, auto& overrideAttr) -> void {
+        auto& effectiveAttr = overrideAttr ? *overrideAttr : defaultAttr;
         const auto& defaultGL = static_cast<const gl::VertexAttributeGL&>(defaultAttr);
         const auto stride = defaultAttr.getStride();
         const auto offset = static_cast<uint32_t>(allData.size());
         const auto index = static_cast<std::size_t>(defaultGL.getIndex());
 
+        bindings.resize(std::max(bindings.size(), index + 1));
+
+        if (const auto& buffer = gl::VertexAttributeGL::getSharedRawBuffer(*this, effectiveAttr, usage)) {
+            bindings[index] = {
+                /*.attribute = */ {effectiveAttr.getSharedType(), effectiveAttr.getSharedOffset()},
```

Comment:
`getSharedOffset` and `getSharedVertexOffset` should return `uint32_t` or a static_cast should be done here. (narrowing conversion warning, MSVC C4838).